pybossa/util.py

```diff
@@ -1207,6 +1214,32 @@ def process_annex_load(tp_code, response_value):
         count += 1
     return tp_code
 
+
 def admin_or_project_owner(user, project):
     if not (user.is_authenticated and (user.admin or user.id in project.owners_ids)):
         raise abort(403)
+
+
+tp_component_tags = ["text-input", "dropdown-input", "radio-group-input",
```

Comment:
I feel it would be convenient to define a variable close to a function who calls it. But if it would be called by multiple places, I would definitely define it at the top.

Reply:
We define all variables and imports at the top and can continue doing same irrespective of number of times its used.

Reply:
It is quite common in modern open source projects defining variables near where they are used. e.g. "[tensorflow](https://github.com/tensorflow/tensorflow/blob/bd054e6bb22e6861521805073e9152fe8a89a3a2/tensorflow/python/util/function_utils.py#L122)" and "[django](https://github.com/django/django/blob/de74a74b4b889c986cabab837bb03e2113880afe/django/utils/timezone.py#L306)".

Reply:
yes, different projects can have different coding convention that they follow.

Reply:
If you have strong preference, I can move it to the top. I just feel defining a variable 1000 lines above is not convenient nor a good practice.